### PR TITLE
Adds skills folder to repo and loads markdown for each skill

### LINKED ISSUE #286: add `/skills` folder to the repo and load that markdown for each skill

### PR DESCRIPTION
**Pull request by lumodon**
Fixes #286 

## Overview
1. We modified the skills digest to include 'markdownFilePath'
1. Changed helper function 'renderMarkdown' & associated helpers to be abstract - used for modules AND skills
1. Created new bin meta-helper for generating folders and initial file markdowns (not useful anymore, keeping until told otherwise)
1. Includes styling for skills markdown view
1. Modified skills view file itself
1. Modified skills route to use 'renderMarkdown' helper

## Configuration Changes
Modifies package.json scripts section - script: `test` to include 'skills' folder in tests
No tests written yet

### CODE REVIEW

**Comment by deadlyicon** [comment]
@lumodon This is very close. I have some small changes I'd like to talk with you and your pair in-person. Lets plan to meet after the sprint planning meeting.

File: bin/rename-skill [hunk 1]
```diff
@@ -29,18 +32,51 @@ console.log(
   chalk.magenta(util.inspect(newSkillNames))
 )
 
-const renameSkillsWithinModules = () =>
```
Comment (lumodon) [inline bin/rename-skill]: rename this to `renameSkills`

File: bin/rename-skill [hunk 2]
```diff
@@ -29,18 +32,51 @@ console.log(
...
+        modules: renameSkillsWithinModule,
+        skills: renameSkillsWithinMarkdown,
+      }[container])
     )
   })
 
+const renameSkillsWithinMarkdown = (skill) => {
+  const oldMarkdownPath = path.join(__dirname, '..', skill.markdownFilePath)
+  const newMarkdownPath = path.join(
+    __dirname,
```
Comment (lumodon) [inline bin/rename-skill]: there is a utility for this in `digest/utils`

File: bin/rename-skill [hunk 3]
```diff
@@ -29,18 +32,51 @@ console.log(
...
+  ) + '/' + newSkillNames[0].replace(/[\'\,\/\s]/g, '-') + '.md'
+
+  return fs.readFile(oldMarkdownPath)
+    .then(buffer => buffer.toString())
+    .then(rawtext => {
+      const newText = rawtext.replace(/\#.*\n/, `# ${newSkillNames[0]}\n`)
+      return fs.writeFile(newMarkdownPath, newText)
+    })
+    .then( () => fs.unlink(oldMarkdownPath))
+    .catch(err => {
```
Comment (lumodon) [inline bin/rename-skill]: just remove this, errors tht propgate up should already halt the rename-skill script

Comment (lumodon) [reply bin/rename-skill]: make sure the output of the script tells you which ones were dont good and not done

File: bin/rename-skill [hunk 4]
```diff
@@ -111,7 +147,8 @@ const yyyymmddhhmmss = () => {
 
 Promise.all([
   generateMigration(),
-  renameSkillsWithinModules(),
+  renameSkillsWithinContainer('modules'),
```
Comment (lumodon) [inline bin/rename-skill]: merge these functions into renameSkill

File: skills/Can-add-.-node_modules-.bin-to-your-$PATH,-in-the-shell.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Can add `./node_modules/.bin` to your `$PATH`, in the shell
```
Comment (lumodon) [inline skills/Can-add-.-node_modules-.bin-to-your-$PATH,-in-the-shell.md]: please space things out 

```markdown
# Can add `./node_modules/.bin` to your `$PATH`, in the shell

_description tbd_

## Resources

### Reading

_nothing here yet, please add something_

### Watching

_nothing here yet, please add something_

```

**Pull request closed without merging**